src/Chapter2.hs

```diff
@@ -841,7 +859,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n xs
+  | n < 0 = []
+  | otherwise = take size $ drop n $ cycle xs
+        where
+            size = length xs
```

Comment:
Nice solution!
As an additional speed-up for a corner-case, you can return the list itself if `n == 0` 🚤 
But your solution already covers all cases, this is just a thought to improve performance in some cases 🙂